shardingsphere-proxy/shardingsphere-proxy-frontend/shardingsphere-proxy-frontend-mysql/src/main/java/org/apache/shardingsphere/proxy/frontend/mysql/command/query/text/query/MySQLComQueryPacketExecutor.java

```diff
@@ -71,10 +80,19 @@ public MySQLComQueryPacketExecutor(final MySQLComQueryPacket packet, final Conne
     }
     
     private Collection<DatabasePacket<?>> processUpdate(final UpdateResponseHeader updateResponseHeader) {
-        responseType = ResponseType.UPDATE;
         return ResponsePacketBuilder.buildUpdateResponsePackets(updateResponseHeader);
     }
     
+    private Collection<DatabasePacket<?>> processClientEncoding(final ClientEncodingResponseHeader clientEncodingResponseHeader) {
+        String value = clientEncodingResponseHeader.getValue();
+        Optional<Charset> charset = MySQLCharacterSet.findByValue(value);
+        if (charset.isPresent()) {
+            clientEncodingResponseHeader.getConnectionSession().getAttributeMap().attr(CommonConstants.CHARSET_ATTRIBUTE_KEY).set(charset.get());
```

Comment:
There is another value `MySQLConstants.MYSQL_CHARACTER_SET_ATTRIBUTE_KEY` which represented MySQL charset and collation is stored in channel attributes map. What if the charset changed to `GBK` while the collation are still `utf8mb4`? The wrong collation id may be returned to client.

Reply:
When using MySQL set character encoding not just for interacting with the client-side protocol. 
The storage of data will also changed.
I did test. The performance of proxy is consistent with MySQL.